pkg/metrics/metrics.go

```diff
@@ -0,0 +1,33 @@
+package metrics
+
+import "github.com/prometheus/client_golang/prometheus"
+
+var (
+	registry                 = prometheus.NewRegistry()
+	storageBackendConfigured = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "image_registry_operator",
+		Subsystem: "storage_backend",
+		Name:      "configured",
+		Help:      "Indicates if registry has configured storage.",
+	})
+	storageBackendReconfigured = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "image_registry_operator",
+		Subsystem: "storage_backend",
+		Name:      "reconfigured",
+		Help:      "Number of times registry storage backend got reconfigured.",
+	})
+	degraded = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "image_registry_operator",
+		Subsystem: "status",
+		Name:      "degraded",
```

Comment:
Perhaps we should include a `reason` label as a part of this metric.

The we can be more intelligent about the alerting:

1. Warning if degraded with `reason=StorageNotConfigured` for 48h
2. Critical otherwise

We may also be able to remove the gauge that merely indicates if storage has been configured.